src/parser/Iri.h

```diff
@@ -36,15 +37,30 @@ class Iri {
   const std::string& toStringRepresentation() const;
   std::string& toStringRepresentation();
 
-  // Create a new `Iri` given an iri string with brackets.
+  // Create a new `Ìri` given an IRI string with brackets.
   static Iri fromIriref(std::string_view stringWithBrackets);
 
-  // Create a new `Iri` given an iri string without brackets.
+  // Create a new `Iri` given an IRI string without brackets.
   static Iri fromIrirefWithoutBrackets(std::string_view stringWithoutBrackets);
 
-  // Create a new iri given a prefix iri and its suffix
+  // Create a new `Iri` given a prefix IRI and its suffix
   static Iri fromPrefixAndSuffix(const Iri& prefix, std::string_view suffix);
 
+  // Create a new `Iri` object, considering the base IRI. For IRIs with a scheme
+  // (like `<http://...>`), this is the same as `fromIriref`. For IRIs without a
+  // scheme, prepend the base prefix for relative IRIs (like `<UPI001AF4585D>`)
+  // or for absolute IRIs (like `</prosite/PS51927>`).
+  static Iri fromIrirefConsiderBase(std::string_view iriStringWithBrackets,
+                                    const Iri& basePrefixForRelativeIris,
+                                    const Iri& basePrefixForAbsoluteIris);
+
+  // Get the base IRI from this `Iri` object. For example, the base IRI of
+  // `<http://purl.uniprot.org/uniprot/>` is `<http://purl.uniprot.org/>`.
+  Iri getBaseIri() const;
+
+  // Return true iff the IRI is empty.
+  bool empty() const { return iri_.empty(); }
```

Comment:
I asked myself the same question. The default constructor for `Iri`, which is public, does exactly this. It creates an `Iri`, where `iri_` is the empty string. This is great for this PR because I need a recognizable `Iri` for the case that the base is not set. So I wouldn't change anything here.